stateproof/worker.go

```diff
@@ -122,6 +122,7 @@ func (spw *Worker) Start() {
 func (spw *Worker) Shutdown() {
 	spw.shutdown()
```

Comment:
Think we should have a unit-test that tries to stop the sp service while the builder and the signer are waiting for a block i.e blocked on `Wait` function. I think it can be done in the worker_test.go